packages/tx/src/baseTransaction.ts

```diff
@@ -40,15 +41,15 @@ interface TransactionCache {
 export abstract class BaseTransaction<TransactionObject> {
   private readonly _type: number
 
-  public readonly nonce: BN
-  public readonly gasLimit: BN
+  public readonly nonce: bigint
+  public readonly gasLimit: bigint
   public readonly to?: Address
-  public readonly value: BN
+  public readonly value: bigint
   public readonly data: Buffer
 
-  public readonly v?: BN
-  public readonly r?: BN
-  public readonly s?: BN
+  public readonly v?: bigint
```

Comment:
A bit of a side topic, but since we change the function signature on this here wanted to mention: thinking for some time now if we want to switch the type of the `v` value to `number` to better differentiate towards `r` and `s` values? No strong opinion here though (but some slight tendency towards doing).